erpnext/selling/doctype/quotation/quotation.js

```diff
@@ -94,10 +96,18 @@ erpnext.selling.QuotationController = class QuotationController extends erpnext.
 	}
 	refresh(doc, dt, dn) {
 		super.refresh(doc, dt, dn);
+		let doctype = "";
+		if (doc.quotation_to == "Customer") {
+			doctype = "Customer";
+		} else if (doc.quotation_to == "Lead") {
+			doctype = "Lead";
+		} else if (doc.quotation_to == "Prospect") {
+			doctype = "Prospect";
+		}
 		frappe.dynamic_link = {
 			doc: this.frm.doc,
 			fieldname: "party_name",
-			doctype: doc.quotation_to == "Customer" ? "Customer" : "Lead",
+			doctype: doctype,
```

Comment:
Why not just do:

`doctype: doc.quotation_to`?

Futureproof.

Reply:
Code snippet uses a series of if and else if statements to explicitly set the variable doctype based on the value of doc.quotation_to. This approach clearly handles each case ("Customer," "Lead," "Prospect") individually, and it's easy to read and modify.

The suggested alternative in your question (doctype: doc.quotation_to == "Customer" ? "Customer" : "Lead") is not equivalent because it only checks if doc.quotation_to is "Customer" and incorrectly defaults to "Lead" without verifying if it should indeed be "Lead." This misses cases like "Prospect" and any other potential values.

Using a ternary operator for this purpose can lead to errors or reduced clarity, especially when multiple conditions are involved. The original method using if-else statements is more appropriate here for clarity and accuracy.

Reply:
Try it.

Reply:
Okay so a filtered Link field rather than Select, but same effect.
I don't need to try it, it can only be these values:
![Screen Shot 2024-04-30 at 18 20 35](https://github.com/frappe/erpnext/assets/110036763/8200fa37-e4fd-4cfe-9c58-48a84934eab0)

Reply:
```javascript
                        frm.set_query("quotation_to", function () {
                                return {
                                        filters: {
                                                name: ["in", ["Customer", "Lead", "Prospect"]],
                                        },
                                };
                        });
```